homeassistant/components/ecovacs/sensor.py

```diff
@@ -344,6 +352,44 @@ async def on_event(event: ErrorEvent) -> None:
         self._subscribe(self._capability.event, on_event)
 
 
+class EcovacsLegacyBatterySensor(EcovacsLegacyEntity, SensorEntity):
+    """Legacy battery sensor."""
+
+    _attr_native_unit_of_measurement = PERCENTAGE
+    _attr_device_class = SensorDeviceClass.BATTERY
+    _attr_entity_category = EntityCategory.DIAGNOSTIC
+
+    def __init__(
+        self,
+        device: VacBot,
+    ) -> None:
+        """Initialize the entity."""
+        super().__init__(device)
+        self._attr_unique_id = f"{device.vacuum['did']}_battery_status"
+
+    async def async_added_to_hass(self) -> None:
+        """Set up the event listeners now that hass is ready."""
+        self._event_listeners.append(
+            self.device.batteryEvents.subscribe(
+                lambda _: self.schedule_update_ha_state()
+            )
+        )
+
+    @property
+    def native_value(self) -> StateType:
+        """Return the value reported by the sensor."""
+        if (status := self.device.battery_status) is not None:
+            return status * 100  # type: ignore[no-any-return]
+        return None
+
+    @property
+    def icon(self) -> str | None:
+        """Return the icon to use in the frontend, if any."""
+        return icon_for_battery_level(
+            battery_level=self.native_value, charging=self.device.is_charging
+        )
```

Comment:
Little bit late :see_no_evil:

but this should not be required anymore after we are supporting range icons. @mib1185 Can you please double check it?